pallets/services-payment/src/lib.rs

```diff
@@ -332,8 +368,12 @@ impl<T: Config> Pallet<T> {
                 WithdrawReasons::FEE,
                 ExistenceRequirement::AllowDeath,
             ) {
-                // Burn for now, we might be able to pass something to do with this
-                drop(imbalance);
+                if let Some(address) = RefundAddress::<T>::get(para_id) {
+                    T::Currency::resolve_creating(&address, imbalance);
+                } else {
+                    // Burn for now, we might be able to pass something to do with this
+                    drop(imbalance);
+                }
             }
         }
```

Comment:
This doesn't clear the refund address storage, add this outside of the if (if we want to clear it):

```
RefundAddress::<T>::remove(para_id);
```

Reply:
good catch, I will probably do a take